samples/emoji-search/ios/app/EmojiSearchApp/IosHostApi.swift

```diff
@@ -0,0 +1,29 @@
+import Foundation
+import shared
+
+class IosHostApi : PresentersHostApi {
+    private let client: URLSession = .init(configuration: .default)
+    
+    func httpCall(url: String, headers: [String : String], completionHandler: @escaping (String?, Error?) -> Void) {
+        var request = URLRequest(url: URL(string: url)!)
+        for (name, value) in headers {
+            request.addValue(value, forHTTPHeaderField: name)
+        }
+        let task = client.dataTask(with: request) { data, response, error in
+            // The KMM memory model doesn't do shared objects well, so Zipline expects the callback
+            // on the same thread that the download was initiated from. This happens to be the main
+            // thread, so we can bounce back to that thread for now.
+            // Switching to the new KMM memory model may remove the need for this.
```

Comment:
Nice call out

Reply:
I think this was copy/pasted from some older code. We should try again now that we're using the new memory model.

Reply:
Yeah I got it from our internal HTTP client